src/main/java/org/jabref/gui/WaitForBackgroundtasksFinishedDialog.java

```diff
@@ -0,0 +1,49 @@
+package org.jabref.gui;
+
+import javafx.concurrent.Task;
+import org.jabref.logic.l10n.Localization;
+
+import java.util.List;
+
+/**
+ * Dialog shown when closing of application needs to wait for some background tasks.
+ */
+public class WaitForBackgroundtasksFinishedDialog {
+
+    private final DialogService dialogService;
+
+    public WaitForBackgroundtasksFinishedDialog(DialogService dialogService) {
+        this.dialogService = dialogService;
+    }
+
+    public void showAndWait(StateManager stateManager) {
+        if (stateManager.anyTaskRunningBinding.getValue()) {
+            Task<Void> waitForBackgroundtasksFinished = new Task<Void>() {
+                @Override
+                protected Void call() throws Exception {
+                    System.out.println("THREAD STARTED");
+                    while (stateManager.anyTaskRunningBinding.getValue()) {
```

Comment:
Just a wild guess, could it be that actually all tasks are finished, because the `TaskExectutor` is already shutdown?

Reply:
I added some logs in JabRefMain's stop method, it is run after the dialogue is already gone.